pre_commit/languages/dotnet.py

```diff
@@ -76,9 +75,14 @@ def install_environment(
             ),
         )
 
-        # Cleanup build output
-        for d in ('bin', 'obj', build_dir):
-            rmtree(prefix.path(d))
+        # Clean the git dir, ignoring the environment dir
+        helpers.run_setup_cmd(
+            prefix,
+            (
+                'git', 'clean', '-fxd',
```

Comment:
I've actually got a video coming up soon that indicates this should be `-ffxd` :)